test/cbmc/sources/core_json_contracts.c

```diff
@@ -39,7 +39,7 @@
 bool isValidBoundedBuffer( char * buf,
                            size_t max )
 {
-    return ( 0 < max && max < CBMC_MAX_BUFSIZE )
+    return ( 0U < max )
```

Comment:
So I know this is already all here, and not expecting you to go and change all of these
But I really dislike this coding style. It's rather hard for a human to follow, which is clear since there needs to be a comment on the top of the file explaining it.

Then I feel like a compiler is going to optimize this
```
    return ( start >= old_start ) &&
           ( ( old_start < max ) ? ( start <= max ) : ( start == old_start ) );
```

To be, for all intents and purposes, the same as
```
bool retVal = false;
if ( ( start >= old start ) && ( old_start < max) )
{
    retVal = ( start <= max );
}
else
{
    retVal = ( start == old_start );
}
return ( retVal);
```

Reply:
Unfortunately, we use this style because the style you suggested leads to drastic performance drawbacks when CBMC's back-end is solving the mathematical formula generated from the code. We can focus on this PR to include the loop invariants and I can certainly open a GitHub issue so we can properly discuss what would be a more readable C-style for these specifications.